### PR TITLE
Introduced DataController#init method.

### PR DESCRIPTION
**Pull request by oskarwrobel**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Introduced decorable `DataController#init` metohd. Closes https://github.com/ckeditor/ckeditor5-core/issues/120.

### CODE REVIEW

File: src/controller/datacontroller.js [hunk 1]
```diff
@@ -169,9 +172,35 @@ export default class DataController {
...
 
+	/**
+	 * Sets initial input data parsed by the {@link #processor data processor} and
+	 * converted by the {@link #upcastDispatcher view-to-model converters}.
+	 * Initial data can be set only to document that {@link module:engine/model/document~Document#version} is equal 0.
+	 *
+	 * **Note** This method is {@link module:utils/observablemixin~ObservableMixin#decorate decorated} which is
+	 * used by e.g. collaborative editing plugin that syncs remote data on init.
+	 *
+	 * @fires set
```
Comment (pjasiun) [inline src/controller/datacontroller.js]: init

File: src/controller/datacontroller.js [hunk 2]
```diff
@@ -169,9 +172,35 @@ export default class DataController {
...
+	 * @param {String} data Input data.
+	 * @param {String} [rootName='main'] Root name.
+	 */
+	init( data, rootName = 'main' ) {
+		if ( this.model.document.version ) {
+			throw new CKEditorError( 'datacontroller-init-document-data-initialized: Trying to set initial data to initialized document.' );
```
Comment (pjasiun) [inline src/controller/datacontroller.js]: Error needs some docs. You can explain there that `init` should be called only once, during document initialization, when there is no content yet to set the initial content. You can not init document later.

Comment (oskarwrobel) [reply src/controller/datacontroller.js]: Done.

File: src/controller/datacontroller.js [hunk 3]
```diff
@@ -169,9 +172,42 @@ export default class DataController {
...
+	 * used by e.g. collaborative editing plugin that syncs remote data on init.
+	 *
+	 * @fires init
+	 * @param {String} data Input data.
+	 * @param {String} [rootName='main'] Root name.
+	 */
+	init( data, rootName = 'main' ) {
+		if ( this.model.document.version ) {
+			/**
+			 * Cannot initialize already initialized data. Data should be initialized only once, during
```
Comment (Reinmar) [inline src/controller/datacontroller.js]: Data isn't the thing that you initialize. It's the editor. You initialize the editor (or model, in engine-oriented context) with data.

Comment (oskarwrobel) [reply src/controller/datacontroller.js]: Changed.

**Pull request merged**